### DISCUSSION TITLE
How can I hide an item in ComboBox?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11371] [Q&A] [answered]
I want to show/hide an item by viewmodel in ComboBox. Here is my code:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication2.MainWindow"
        Title="AvaloniaApplication2" Name="Root">
	<ComboBox x:Name="PositionCB"  Width="150" HorizontalAlignment="Center" Height="27" Padding="0" VerticalAlignment="Center" >
		<ComboBox.ItemTemplate>
			<DataTemplate>
				<TextBlock Text="{Binding Name}" IsVisible="{Binding IsVisible}" VerticalAlignment="Center"></TextBlock>
			</DataTemplate>
		</ComboBox.ItemTemplate>
	</ComboBox>
</Window>
```

```
using Avalonia.Controls;
using Avalonia.Data.Converters;
using System;
using System.Collections.Generic;
using System.Globalization;

namespace AvaloniaApplication2
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
            PositionCB.ItemsSource=new List<TestModel>() { new TestModel() { Name="1",IsVisible=true}, new TestModel() { Name = "2", IsVisible = false }, new TestModel() { Name = "3", IsVisible = true } };
        }
        public class TestModel {
            public string Name { get; set; } = "";
            public Boolean IsVisible { get; set; } = false;
        }
    }
  
}
```

And now as you see, in spite of the second one not visible any more, there still has a blank option here.
![1](https://github.com/AvaloniaUI/Avalonia/assets/30720036/1ad54f22-55a9-422d-abe9-60801733c52d)

How can I solve this?

**Comment by tlitookilakin** [comment]
Take a look at the template for combobox items in whatever theme you're using. Your Textblock is getting wrapped in something else, which is adding padding, and you can use a Style to remove that padding.

**Comment by timunie** [comment, marked as answer]
I'd suggest to bind to a filtered list instead tbh. DynamicData (which is shipped with ReactiveUI, but doesn't need ReactiveUI to work) can be very handy here. Check out: https://dynamic-data.org

**Reply by kyuranger** [reply]
> I'd suggest to bind to a filtered list instead tbh. DynamicData (which is shipped with ReactiveUI, but doesn't need ReactiveUI to work) can be very handy here. Check out: https://dynamic-data.org

Thank you so much. I like this answer so much!
